### ISSUE TITLE
job.wait returns None when already closed instead of real result

### ISSUE THREAD

**Issue by smagafurov** [aio-libs/aiojobs#119]
```python
import asyncio
import aiojobs


async def func():
    return 123


async def probe():
    scheduler = await aiojobs.create_scheduler()

    job_list = []
    for i in range(10):
        job = await scheduler.spawn(func())
        job_list.append(job)

    res_list = []
    for job in job_list:
        res = await job.wait()
        res_list.append(res)

    print(res_list)

    await scheduler.close()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(probe())
```

Expected it prints:
```python
[123, 123, 123, 123, 123, 123, 123, 123, 123, 123]
```

But It prints:

```python
[123, None, None, None, None, None, None, None, None, None]
```

aiojobs-0.2.2

**Comment by smagafurov** [comment]
Workaround:

```python
import asyncio
import aiojobs


async def func():
    return 123


async def probe():
    scheduler = await aiojobs.create_scheduler()

    job_list = []
    for i in range(10):
        job = await scheduler.spawn(func())
        job._explicit = True  # <------------ workaround step 1
        job_list.append(job)

    res_list = []
    for job in job_list:
        res = await job._do_wait(timeout=None)  # <------------ workaround step 2
        res_list.append(res)

    print(res_list)

    await scheduler.close()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(probe())
```

**Comment by pyhedgehog** [comment]
Slightly better workaround:
```python
...
    res_list = []
    for job in job_list:
        await job.wait()
        res = job._taks.result()
        res_list.append(res)
...
```
This variant at least respects `scheduler` parameter `limit`.

**Comment by Dreamsorcerer** [comment]
Fixed in #343.
